---
title: "SKCM Driver Genes Detection"
author: "Sofia Gamino"
date: "2025-05-15"
output: html_document
---

# Detection of drivers in Skin Cancer Melanoma

> You need the package dndscv to run this code.

## Generating the input file

With the mutation data of SKCM available from the TCGA (`SKCM/data_mutations.txt`) we can retrieve the information needed for the generation of the dndscv object.

We are going to take just 5 columns, containing the following: 1) Sample ID
2) Chromosome 3) Position 4) Reference 5) Mutation.

```{r}
library(dndscv)
SKCM_muts_raw = read.table("SKCM/data_mutations.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)
SKCM_muts = SKCM_muts_raw[,c("Tumor_Sample_Barcode", "Chromosome", "Start_Position", "Reference_Allele", "Tumor_Seq_Allele2")]
```

### Detection of hypermutators

Hypermutators are samples that suffer from a great burden of mutations. Sometimes, these mutations are under mutational procceses that cannot be modeled with a trinucleotide-substitution model. However, as the skin is the organ that resists the most mutational burden of all organs, it is expected to find several hypermutators, therefore, iá¸¿ going to rise the threshold for the definition of a hypermutator, with samples with more than 760 mutations  (>20 muts per mmegabase) considered as hypermutators (https://www.frontiersin.org/journals/immunology/articles/10.3389/fimmu.2022.1006665/full)

```{r}
tumor_muts = table(SKCM_muts$Tumor_Sample_Barcode)
(Hypermuts = tumor_muts[tumor_muts > 760])
```
Now, we have 108 samples classified as hypermutators. 

## Driver Detection

If you want to check drivers including the hypermutators, you can change the maximum of muts per gene per sample to 10 and the max of coding muts per sample to 10,000. This though, will mess with the statistical power of dndscv.

```{r}
SKCM_dndscv_hyp = dndscv(SKCM_muts, max_muts_per_gene_per_sample = 10, max_coding_muts_per_sample = 5000, outmats = TRUE)
SKCM_dndscv_non_hyp = dndscv(SKCM_muts, max_muts_per_gene_per_sample = 3, max_coding_muts_per_sample = 760, outmats = TRUE)

```

Now we need to filter those mutations who have significant q-values. We will choose only those genes

```{r}
SKCM_drivers_hyp = SKCM_dndscv_hyp$sel_cv[which(SKCM_dndscv_hyp$sel_cv$qglobal_cv < 0.1 & SKCM_dndscv_hyp$sel_cv$n_mis > 0 & SKCM_dndscv_hyp$sel_cv$n_non > 0), ]
SKCM_drivers_non_hyp = SKCM_dndscv_non_hyp$sel_cv[which(SKCM_dndscv_non_hyp$sel_cv$qglobal_cv < 0.1 & SKCM_dndscv_non_hyp$sel_cv$n_mis > 0 & SKCM_dndscv_non_hyp$sel_cv$n_non > 0), ]
SKCM_drivers_hyp_fil = SKCM_drivers_hyp[order(-SKCM_drivers_hyp$wmis_cv),]
SKCM_drivers_non_hyp_fil = SKCM_drivers_non_hyp[order(-SKCM_drivers_non_hyp$wmis_cv),]
(SKCM_drivers_final = union(SKCM_drivers_non_hyp_fil$gene_name, SKCM_drivers_hyp_fil$gene_name))
```


Now that we have a list with our driver genes from hypermutated and non-hypermutated samples, we can extract the mutations present in for each gene and put them in the Cancer Genome interpreter to validate them with existing literature. 

```{r}
library(maftools)
SKCM_maf = read.maf("SKCM/data_mutations.txt")
```

```{r}
SKCM_driver_muts = SKCM_maf@data[SKCM_maf@data$Hugo_Symbol %in% SKCM_drivers_final]
CGI_format = SKCM_driver_muts[, c("Hugo_Symbol", "Tumor_Sample_Barcode", "Chromosome", "Start_Position", "Reference_Allele", "Tumor_Seq_Allele2")]
write.table(CGI_format, "Total_muts_q0.1_CGI.tsv", sep = "\t", row.names = FALSE, quote = FALSE)
```


